### PR TITLE
Feature 641 mtd feature relative

### LINKED ISSUE #641: New Use Case: Feature Relative using MTD output for feature centroid lat/lon
## Describe the Enhancement ##
Currently the Feature Relative use-cases identify the location of the feature from output from the GFDL tracker.  The SBU and UW-CIMSS projects would also like to have support for MTD centroid lat/lon.  I think we need to discuss if this can be handled in the python script that provides lat/lons to the script that cuts out the tile prior to passing into Series-Analysis, or if it requires modifications  to TC-pairs/TC-stat to allow for further refinement.    

### Time Estimate ###
*Estimate the amount of work required here.*
*Issues should represent approximately 1 to 3 days of work.*

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
December 2020

### Funding Source ###

7791041

### Assignee ###
- [x] Select **engineer(s)** or **no engineer** required
- [x] Select **scientist(s)** or **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [x] Select **milestone** to next major version milestone or "Future Versions"

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [x] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [x] Complete the development and test your changes.
- [x] Add/update log messages for easier debugging.
- [x] Add/update unit tests.
- [x] Add/update documentation.
- [x] Push local changes to GitHub.
- [x] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [x] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, **Milestone**, and **Linked issues**
- [x] Iterate until the reviewer(s) accept and merge your changes.
- [x] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by DanielAdriaansen**
Closes #641. Also includes bug fix to ExtractTiles wrapper for MTD support.

- [x] Describe testing already performed for these changes:</br>
Ran locally on kiowa, and inspected output from MTD and ExtractTiles in ncview, and reviewed PNG files created by SeriesAnalysis.

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Review documentation, and METplus conf file.

- [ ] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [ ] Do these changes include sufficient testing updates? **[Yes]**

- [ ] Will this PR result in changes to the test suite? **[Yes]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>
New sample data and new use case test.

- [ ] Please complete this pull request review by **[ASAP]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

**Comment by DanielAdriaansen** [comment]
@TaraJensen, I ended up using PWAT from some existing GFS data.

@georgemccabe, the GitHub actions new=True needs to be changed prior to merging, I have not changed that yet.

**Comment by DanielAdriaansen** [comment]
ToDo:

- [x] Add useful gallery image to documentation
- [x] Remove unused lines from METplus conf file
- [x] Double check use case output section in documentation

**Comment by DanielAdriaansen** [comment]
@georgemccabe I still did not turn new=True back to False in the json file, but I don't want to do that now since it will trigger automation again so maybe we can do it just prior to merging? Everything else is done and ready to go though.

**Comment by georgemccabe** [comment]
@DanielAdriaansen, I am working on resolving the failing use cases due to a cartopy issue (see discussion in PR #1152 for more details). Once that is finished I need to update the truth data because truth data for 2 use cases need to be updated. Once all of that is done, we can merge in the changes from develop, update the new use case group so run = false, and push to trigger the tests again.

**Comment by georgemccabe** [comment]
Use case documentation can be found here:
https://metplus.readthedocs.io/en/feature_641_mtd_feature_relative/generated/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.html#sphx-glr-generated-model-applications-medium-range-mtd-seriesanalysis-fcstgfs-obsgfs-featurerelative-seriesbylead-py

File: docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+#  * Series-Analysis Observation dataset: GFS Analysis
+#
+
+##############################################################################
+# METplus Components
+# ------------------
+#
+# This use case first runs MODE Time Domain and ExtractTiles wrappers to
+# generate tiles of data centered on objects defined using MTD. The MET
+# regrid-dataplane tool is used to regrid the data (GRIB1 or GRIB2 into netCDF).
```
Comment (georgemccabe) [inline docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py]:
```suggestion
# regrid_data_plane tool is used to regrid the data (GRIB1 or GRIB2 into netCDF).
```

File: docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+# |
+#
+
+##############################################################################
+# METplus Configuration
+# ---------------------
+#
+# METplus first loads all of the configuration files found in parm/metplus_config,
+# then it loads any configuration files passed to METplus via the command line
+# with the -c option, i.e. -c parm/use_cases/model_applications/medium_range/TCStat_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.conf
```
Comment (georgemccabe) [inline docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py]:
```suggestion
```
-c option is no longer needed

File: docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+# | **Forecast lead:** 6, 12, 18, 24, 30
+# |
+#
+
+##############################################################################
+# METplus Configuration
+# ---------------------
+#
+# METplus first loads all of the configuration files found in parm/metplus_config,
+# then it loads any configuration files passed to METplus via the command line
```
Comment (georgemccabe) [inline docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py]:
```suggestion
# then it loads any configuration files passed to METplus via the command line.
```

File: docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+##############################################################################
+# Running METplus
+# ---------------
+#
+# This use case can be run two ways:
+#
+# 1) Passing in MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.conf, then a user-specific system configuration file::
+#
+#        run_metplus.py -c /path/to/METplus/parm/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.conf
```
Comment (georgemccabe) [inline docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py]:
```suggestion
#        run_metplus.py /path/to/METplus/parm/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.conf
```

File: docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+#        -c /path/to/user_system.conf
```
Comment (georgemccabe) [inline docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py]:
```suggestion
#        /path/to/user_system.conf
```

File: docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+#
+# 2) Modifying the configurations in parm/metplus_config, then passing in MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.conf::
+#
+#        run_metplus.py -c /path/to/METplus/parm/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.conf
```
Comment (georgemccabe) [inline docs/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.py]:
```suggestion
#        run_metplus.py -/path/to/METplus/parm/use_cases/model_applications/medium_range/MTD_SeriesAnalysis_fcstGFS_obsGFS_FeatureRelative_SeriesByLead.conf
```

**Review by georgemccabe — APPROVED** [review]
I reviewed the documentation file and cleaned up a few things that were out-dated. I also rearranged the config file for better readability.

In this PR, a fix to ExtractTiles wrapper was needed because an incorrect assumption about the MTD output was made based on the data used in the previous ExtractTiles using MTD input use case. Dan also renamed a variable used in the Contributor's Guide instructions for adding data for a new use case so that the steps are more clear.

The changes to the config file triggered the automated tests again, so given that the use case runs without error, I approve of this PR.

Note that after this PR is merged into develop, the develop-ref branch should be updated so that the truth data includes the new output generated from this new use case.

**Pull request merged**